### PR TITLE
[W4.5c][W13-A2] Aadyaa Maddi

### PR DESCRIPTION
**Pull request by amad-person**
Handled exception that was thrown when the data file was set as read-only.

### CODE REVIEW

File: src/seedu/addressbook/Main.java [hunk 1]
```diff
@@ -105,16 +107,21 @@ private void recordResult(CommandResult result) {
...
      * @return result of the command
      */
-    private CommandResult executeCommand(Command command)  {
+    private CommandResult executeCommand(Command command) {
         try {
             command.setData(addressBook, lastShownList);
             CommandResult result = command.execute();
             storage.save(addressBook);
             return result;
+        } catch (StorageOperationException soe) {
```
Comment (okkhoy) [inline src/seedu/addressbook/Main.java:116]: In general, it is good to handle the exception closer to where it occurs.

File: src/seedu/addressbook/Main.java [hunk 2]
```diff
@@ -105,16 +107,21 @@ private void recordResult(CommandResult result) {
...
+            ui.showToUser(MESSAGE_READ_ONLY);
+            start(new String[0]);
```
Comment (okkhoy) [inline src/seedu/addressbook/Main.java:118]: This is one of the ways you can handle the exception. 
Think how you can improve the user experience in light of the exception that just occurred, and not lose the data.

**Comment by okkhoy** [comment]
@amad-person 
Some comments added. Please ack & close the PR after reading comments.

**Comment by amad-person** [comment]
Acknowledged!

**Pull request closed without merging**